src/aat/java/uk/gov/hmcts/reform/coh/functional/bdd/steps/ApiSteps.java

```diff
@@ -101,7 +109,15 @@ public void cleanUp() {
                 log.error("Failure may be due to foreign key. This is okay because the online hearing will be deleted elsewhere.");
             }
         }
-
+        if(testContext.getScenarioContext().getSessionEventForwardingRegisters() != null) {
+            for (SessionEventForwardingRegister sessionEventForwardingRegister : testContext.getScenarioContext().getSessionEventForwardingRegisters()) {
+                try {
+                    sessionEventForwardingRegisterRepository.delete(sessionEventForwardingRegister);
+                } catch (DataIntegrityViolationException e) {
+                    log.error("Failure may be due to foreign key. This is okay because the online hearing will be deleted elsewhere.");
+                }
+            }
+        }
```

Comment:
Because in some question round tests the Jurisdiction is created & deleted during the cucumber tests. If we leave the register entry and attempt to delete the generated jurisdiction we will get an error.

Reply:
Why does that affect the register?

Reply:
The register composite key uses the jurisdiction key & event type key.

Reply:
Ah okay. I didn't notice there was a cucumber test to insert a dummy jurisdiction.